### DISCUSSION TITLE
Drag & Drop of blocks into other blocks or migration of blocks from one to another

### DISCUSSION THREAD

**Discussion by wiesson** [payloadcms/payload#2977] [Feature Requests & Ideas]
We often have the case that we would like to re-arrange sections but each block has a container block. At the moment, we have to manually create a new block and copy and paste everything manually but it would be nice if the block could be moved via drag & drop (or some dropdown) to quickly rearrange them.

**Comment by christian-reichart** [comment]
+1 for this.

Idea here:
Maybe something that would be easier to implement then a complex drag & drop UI would be to copy & paste a block.

So you could hit "copy" on any block and then be able to paste inside of any block area (it has to check if the block is allowed there though)

**Comment by dacgray** [comment]
Massive +1 for this:

We nest blocks.  In this case we would need to re-create `My Cool Banner` in `Inner Box 2`.

![image](https://github.com/payloadcms/payload/assets/12794616/4d416ecc-1d1c-4b4e-a816-649f6cb0ab18)

This is a massive pain point when moving lots of detailed blocks.  If we want to move blocks in the case case below we would need to manually re-create all About Us content.

![image](https://github.com/payloadcms/payload/assets/12794616/7a7c7a4b-5156-4ac1-9644-6836520d0171)

IMHO A copy paste function would be best.  This would let us paste blocks between collections too.

**Comment by dacgray** [comment]
Something like this would be fantastic:

![image](https://github.com/payloadcms/payload/assets/12794616/97e64474-e92f-46b7-923c-f0b69a4e2a2b)

**Reply by matthijs166** [reply]
Oehh cool! I think that the past button should be under the Copy button in the dropdown

**Comment by JustusCF** [comment]
+1 from my side, too. This would help so much!

**Comment by alexcrazyfactory** [comment]
+1 really helpful!

**Comment by jakobpevec** [comment]
+1 would speed up the editing process a lot

**Comment by Girbi** [comment]
Please make this posible somehow!!

**Comment by ndefie** [comment]
+1 on this

**Comment by codeflorist** [comment]
Such a feature would indeed be great and is somewhat common in other CMSs.

I would like to add, that copy/pasting should also work across documents. Say you have made a certain hero-section and want to re-use that on other pages. I guess that is a rather common usecase.

Ideally this should also work for items of an array-field (maybe even with multi-select), and not only for blocks.

**Comment by jhb-dev** [comment]
related PR: https://github.com/payloadcms/payload/pull/11513

**Comment by azeller** [comment]
Hey everyone, @jhb-dev already provided the copy/paste functionality and it has been merged a while ago. Now drag and drop should just be an abstraction of copy/paste/delete - right?

It's hard to imagine someone hasn't provided this already? This feature request is almost three years old and the core code for this functionality is already available. Would anybody be willing to build this? Estimations are welcome, we're looking to have this in our systems as soon as possible. If some dev offers this feature development for a good price, we'd of course open source it and have it merged back to the payload core.
